src/passkey_child/passkey_child_assert.c

```diff
@@ -0,0 +1,270 @@
+    if (ret != EOK) {
+        goto done;
+    }
+
+done:
+    talloc_free(tmp_ctx);
+
+    return ret;
+}
+
+errno_t
+reset_public_key(struct passkey_data *data)
+{
+    if (data->type == COSE_ES256) {
+        es256_pk_free((es256_pk_t **) &data->public_key);
+    } else if (data->type == COSE_RS256) {
+        rs256_pk_free((rs256_pk_t **) &data->public_key);
+    } else if (data->type == COSE_EDDSA) {
+        eddsa_pk_free((eddsa_pk_t **) &data->public_key);
+    }
+
+    return EOK;
+}
+
+errno_t
+decode_public_key(struct passkey_data *data)
+{
+    TALLOC_CTX *tmp_ctx = NULL;
+    unsigned char *buf = NULL;
+    size_t buf_len;
+    errno_t ret;
+
+    tmp_ctx = talloc_new(NULL);
+    if (tmp_ctx == NULL) {
+        ret = ENOMEM;
+        goto done;
+    }
+
+    buf = sss_base64_decode(tmp_ctx, data->b64_public_key, &buf_len);
+    if (buf == NULL) {
+        DEBUG(SSSDBG_OP_FAILURE, "Failed to decode public key.\n");
+        ret = ENOMEM;
+        goto done;
+    }
+
+    if (data->type == COSE_ES256) {
+        data->public_key = es256_pk_new();
+        if (data->public_key == NULL) {
+            DEBUG(SSSDBG_OP_FAILURE, "es256_pk_new failed.\n");
+            ret = ENOMEM;
+            goto done;
+        }
+
+        ret = es256_pk_from_ptr(data->public_key, buf, buf_len);
+        if (ret != FIDO_OK) {
+            DEBUG(SSSDBG_OP_FAILURE,
+                  "es256_pk_from_ptr failed [%d]: %s.\n",
+                  ret, fido_strerr(ret));
+            goto done;
+        }
+    } else if (data->type == COSE_RS256) {
+        data->public_key = rs256_pk_new();
+        if (data->public_key == NULL) {
+            DEBUG(SSSDBG_OP_FAILURE, "rs256_pk_new failed.\n");
+            ret = ENOMEM;
+            goto done;
+        }
+
+        ret = rs256_pk_from_ptr(data->public_key, buf, buf_len);
+        if (ret != FIDO_OK) {
+            DEBUG(SSSDBG_OP_FAILURE,
+                  "rs256_pk_from_ptr failed [%d]: %s.\n",
+                  ret, fido_strerr(ret));
+            goto done;
+        }
+    } else if (data->type == COSE_EDDSA) {
+        data->public_key = eddsa_pk_new();
+        if (data->public_key == NULL) {
+            DEBUG(SSSDBG_OP_FAILURE, "eddsa_pk_new failed.\n");
+            ret = ENOMEM;
+            goto done;
+        }
+
+        ret = eddsa_pk_from_ptr(data->public_key, buf, buf_len);
+        if (ret != FIDO_OK) {
+            DEBUG(SSSDBG_OP_FAILURE,
+                  "eddsa_pk_from_ptr failed [%d]: %s.\n",
+                  ret, fido_strerr(ret));
+            goto done;
+        }
+    }
+
+    ret = EOK;
+
+done:
+    talloc_free(tmp_ctx);
+
+    return ret;
+}
+
+errno_t
+verify_assert(struct passkey_data *data, fido_dev_t *dev,
+              fido_assert_t *assert)
+{
+    char *pin = NULL;
+    bool has_pin;
+    ssize_t pin_len = 0;
+    errno_t ret;
+
+    has_pin = fido_dev_has_pin(dev);
+    if (has_pin == true) {
+        pin_len = read_pin(&pin);
```

Comment:
For my knowledge, how will this be changed when SSSD handles the prompting? just this block of code will be removed?

Reply:
I've removed it and updated the instructions to take into account that the PIN is an input to the child process